drivers/watchdog/wdt_mchp_xec.c

```diff
@@ -0,0 +1,188 @@
+/* wdt_xec.c - Microchip XEC watchdog driver */
+
+/*
+ * Copyright (c) 2019 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_LEVEL CONFIG_WDT_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(wdt_mchp_xec);
+
+#include <drivers/watchdog.h>
+#include <soc.h>
+#include <errno.h>
+#include <assert.h>
+
+#include "wdt_mchp_xec.h"
+
+#define MEC_WDT_INT_SRC_VAL	BIT(2)
+
+#define WDT_XEC_REG_BASE				\
+	((volatile struct wdt_xec_registers *)		\
+	 (DT_INST_0_MICROCHIP_XEC_WATCHDOG_BASE_ADDRESS))
+
+DEVICE_DECLARE(wdt_xec);
+
+static int wdt_xec_setup(struct device *dev, u8_t options)
+{
+	volatile struct wdt_xec_registers *wdt_regs = WDT_XEC_REG_BASE;
+	struct wdt_xec_data *data = dev->driver_data;
+
+	if (wdt_regs->control_reg.wdt_enable) {
+		return -EBUSY;
+	}
+
+	if (!data->timeout_installed) {
+		LOG_ERR("No valid WDT timeout installed");
+		return -EINVAL;
+	}
+
+	if (options & WDT_OPT_PAUSE_IN_SLEEP) {
+		LOG_WRN("WDT_OPT_PAUSE_IN_SLEEP is not supported");
+		return -ENOTSUP;
+	}
+
+	if (options & WDT_OPT_PAUSE_HALTED_BY_DBG) {
+		wdt_regs->control_reg.jtag_stall = 1;
+	}
+
+	wdt_regs->control_reg.wdt_enable = 1;
+
+	LOG_DBG("WDT Setup and enabled");
+
+	return 0;
+}
+
+static int wdt_xec_disable(struct device *dev)
+{
+	volatile struct wdt_xec_registers *wdt_regs = WDT_XEC_REG_BASE;
+	struct wdt_xec_data *data = dev->driver_data;
+
+	if (!wdt_regs->control_reg.wdt_enable) {
+		return -EALREADY;
+	}
+
+	wdt_regs->control_reg.wdt_enable = 0;
+	data->timeout_installed = false;
+
+	LOG_DBG("WDT Disabled");
+
+	return 0;
+}
+
+static int wdt_xec_install_timeout(struct device *dev,
+				   const struct wdt_timeout_cfg *config)
+{
+	volatile struct wdt_xec_registers *wdt_regs = WDT_XEC_REG_BASE;
+	struct wdt_xec_data *data = dev->driver_data;
+
+	if (wdt_regs->control_reg.wdt_enable) {
+		return -EBUSY;
+	}
+
+	if (config->window.min > 0U) {
```

Comment:
Per APi defintion, shouldn't this be -EINVAL
Also, next line could results in disabling current timeout if an invalid parameter is used

Reply:
Indeed, about the -EINVAL.

There is no issue with the line below (besides maybe being superfluous): you cannot install a timeout if the wdt is already running. So if you want to get rid of current timeout, wdt_disable() should be called first.